### DISCUSSION TITLE
zc_install: sql/plugins directory not used on upgrade

### DISCUSSION THREAD

**Discussion by lat9** [zencart/zencart#5830] [Ideas]
For the zc_install processing, I've placed the site's additional SQL changes in the `zc_install/sql/plugins` sub-directory ... but they appear to be included _only_ on an initial install.

**Comment by drbyte** [comment]
That is correct.

The thinking was: it's reasonable to parse the provided files on initial install, because it's clear that they're for install-time-only.

I suppose a naming convention for upgrade-only could be used. But then there are various steps of upgrades for each plugin. I guess if we sort filenames by alpha then there's a bit of dependency mgmt there.

**Comment by lat9** [comment]
Agreed on the naming convention.  What I was _attempting_ to do was consolidate a plethora of database updates required for an upgrade that's in-progress (removing database references for plugins no longer used and the like).

All I needed was a way to get those changes pulled in (just one .sql file) to be run as part of that site-wide update.  If the `zc_install` process simply loaded all files in `/zc_install/plugins/updates` alphabetically, that would provide some level of extend-ability.

**Comment by drbyte** [comment]
The plugin infrastructure in 1.5.8 should handle some form of upgrade handling.

**Comment by scottcwilson** [comment]
Can this be closed?  I thought you could use the SQL Installer Classes in the updated version if changes were required - no?

https://docs.zen-cart.com/dev/plugins/encapsulated_plugins/sql_installation/#sql-installer-classes
